duneana/SolarNuAna/AdjHitsUtils.cc

```diff
@@ -0,0 +1,155 @@
+
+      while (LastSize != NewSize)
+      {
+        std::vector<int> AddNow;
+        for (size_t aL = 0; aL < AdjHitVec.size(); ++aL)
+        {
+          for (size_t nL = 0; nL < MyVec.size(); ++nL)
+          {
+            if (HeavDebug)
+            {
+              std::cerr << "\t\tLooping though AdjVec " << aL << " and  MyVec " << nL
+                        << " AdjHitVec - " << AdjHitVec[aL].Channel() << " & " << AdjHitVec[aL].PeakTime()
+                        << " MVec - " << MyVec[nL].Channel() << " & " << MyVec[nL].PeakTime()
+                        << " Channel " << abs((int)AdjHitVec[aL].Channel() - (int)MyVec[nL].Channel()) << " bool " << (bool)(abs((int)AdjHitVec[aL].Channel() - (int)MyVec[nL].Channel()) <= ChanRange)
+                        << " Time " << abs(AdjHitVec[aL].PeakTime() - MyVec[nL].PeakTime()) << " bool " << (bool)(abs((double)AdjHitVec[aL].PeakTime() - (double)MyVec[nL].PeakTime()) <= TimeRange)
+                        << std::endl;
+            }
+
+            if (abs((int)AdjHitVec[aL].Channel() - (int)MyVec[nL].Channel()) <= ChanRange &&
+                abs((double)AdjHitVec[aL].PeakTime() - (double)MyVec[nL].PeakTime()) <= TimeRange)
+            {
+
+              if (HeavDebug)
+                std::cerr << "\t\t\tFound a new thing!!!" << std::endl;
+              // --- Check that this element isn't already in AddNow.
+              bool AlreadyPres = false;
+
+              for (size_t zz = 0; zz < AddNow.size(); ++zz)
+              {
+                if (AddNow[zz] == (int)nL)
+                  AlreadyPres = true;
+              }
+
+              if (!AlreadyPres)
+                AddNow.push_back(nL);
+            } // If this TPCHit is within the window around one of my other hits.
+          }   // Loop through my vector of colleciton plane hits.
+        }     // Loop through AdjHitVec
+
+        // --- Now loop through AddNow and remove from Marley whilst adding to AdjHitVec
+        std::sort(AddNow.begin(), AddNow.end());
+        for (size_t aa = 0; aa < AddNow.size(); ++aa)
+        {
+          if (HeavDebug)
+          {
+            std::cerr << "\tRemoving element " << AddNow.size() - 1 - aa << " from MyVec ===> "
+                      << MyVec[AddNow[AddNow.size() - 1 - aa]].Channel() << " & " << MyVec[AddNow[AddNow.size() - 1 - aa]].PeakTime()
+                      << std::endl;
+          }
+
+          AdjHitVec.push_back(MyVec[AddNow[AddNow.size() - 1 - aa]]);
+          MyVec.erase(MyVec.begin() + AddNow[AddNow.size() - 1 - aa]); // This line creates segmentation fault
+                                                                      // std::cout << "Erase works" << std::endl;
+        }
+
+        LastSize = NewSize;
+        NewSize = AdjHitVec.size();
+        if (HeavDebug)
+        {
+          std::cerr << "\t---After that pass, AddNow was size " << AddNow.size() << " ==> LastSize is " << LastSize << ", and NewSize is " << NewSize
```

Comment:
Why the mixture of cout and cerr?  You have to look in two places to find your output, and you may lose the ordering of the messages, even if they are combined together e.g. with bash's 2>&1.  I don't know; I haven't tried, but due to output file buffering, I'd imagine you'd get an arbitrary mixture.